modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/RankFeatureFieldMapper.java

```diff
@@ -113,6 +113,19 @@ public Query existsQuery(QueryShardContext context) {
             throw new IllegalArgumentException("[rank_feature] fields do not support sorting, scripting or aggregating");
         }
 
+        @Override
+        public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
+            if (format != null) {
+                throw new IllegalArgumentException("Field [" + name() + "] of type [" + typeName() + "] doesn't support formats.");
+            }
+            return new SourceValueFetcher(name(), mapperService, false) {
```

Comment:
One other change I noticed -- previously we always passed the result of `FieldMapper#parsesArrayValue` instead of hard-coding the booleans. Now there is nothing really tying this to the `parsesArrayValue` and it could be easily overlooked. I wonder how we could make this more robust.